src/error_handling.c

```diff
@@ -25,6 +25,7 @@ extern "C"
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
+#define __STDC_WANT_LIB_EXT1__ 1
```

Comment:
is this required to define here? it looks like `rcutils/error_handling.h` already has defined that.

Reply:
[this example](https://en.cppreference.com/w/c/string/byte/strlen) says that the `define` shall be before the `#include <string.h>`. I'm not sure whether it is good to rely on what was defined in other headers.

````bash
#define __STDC_WANT_LIB_EXT1__ 1
#include <string.h>
#include <stdio.h>
 
int main(void)
{
    const char str[] = "How many characters does this string contain?";
 
    printf("without null character: %zu\n", strlen(str));
    printf("with null character:    %zu\n", sizeof str);
 
#ifdef __STDC_LIB_EXT1__
    printf("without null character: %zu\n", strnlen_s(str, sizeof str));
#endif
}
````